File: backend/app/domain/product/product.py
from dataclasses import dataclass

from domain.common.base_data_field import BaseDataField
from domain.product.gender_values import Gender


@dataclass
class BaseProduct:
    id: int
    name: str
    description: str
    price: int


@dataclass
class CatalogProduct(BaseProduct):
    image: str
    sizes: list[str]


@dataclass
class Product(BaseProduct):
    code: int
    article: str
    images: list[str]
    category: BaseDataField
    brand: BaseDataField
    color: BaseDataField
    sizes: list[str]
    gender: Gender | None = Gender.unisex


@dataclass
class ProductImage:
    image: str

File: frontend/src/app/models/product.ts
import { TypeDataField } from "./filter"


export type TypeBaseProduct = {
    id: number,
    name: string,
    description: string,
    price: number,
}

export interface ICatalogProduct extends TypeBaseProduct {
    image: string,
    sizes: string[]
}

export interface IProduct extends TypeBaseProduct {
    images: string[],
    gender: string,
    category: TypeDataField,
    color: string,
    sizes: string[],
    brand: TypeDataField,
    code: string,
    article: string,
}

export interface ICartProduct extends TypeBaseProduct {
    image: string,
    size: string,
    quantity: number
}
